### PR TITLE
[stable/datadog] Fix trace agent and useConfigMap

### PR DESCRIPTION
**Pull request by ogmaresca**
#### Is this a new chart
No

#### What this PR does / why we need it:

Version 2.0.0 of the stable/datadog introduced new issues.

1. If `agents.useConfigMap` is set to to true, it causes a templating error because `.Values.image` no longer exists.

```
Error: template: datadog/templates/datadog-yaml-configmap.yaml:38:28: executing "datadog/templates/datadog-yaml-configmap.yaml" at <.Values.image.tag>: nil pointer evaluating interface {}.tag
```

2. If `agents.useConfigMap` is false and `agents.apm.enabled` is true, the Datadog trace agent container fails to startup with this error:

```
trace-agent not enabled. Set the environment variable DD_APM_ENABLED=true or add "apm_config.enabled: true" entry to your datadog.yaml. Exiting...
```

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by MichielDeMey** [comment]
Just ran into this issue installing the new `v2.0.1` Helm chart.

Not using the config maps and gives the error
```
trace-agent not enabled. Set the environment variable DD_APM_ENABLED=true or add "apm_config.enabled: true" entry to your datadog.yaml. Exiting...
```

File: stable/datadog/templates/container-trace-agent.yaml [hunk 1]
```diff
@@ -14,6 +14,8 @@
     {{- include "containers-common-env" . | nindent 4 }}
     - name: DD_LOG_LEVEL
       value: {{ .Values.agents.containers.traceAgent.logLevel | default .Values.datadog.logLevel | quote }}
+    - name: DD_APM_ENABLED
+      value: "true"
```
Comment (clamoriniere) [inline stable/datadog/templates/container-trace-agent.yaml]: I think this is the default value already in the configuration

File: stable/datadog/templates/datadog-yaml-configmap.yaml [hunk 2]
```diff
@@ -35,7 +35,7 @@ data:
       max_memory: 0
       max_cpu_percent: 0
 
-    {{- $version := (.Values.image.tag | toString | trimSuffix "-jmx") }}
+    {{- $version := (.Values.agents.image.tag | toString | trimSuffix "-jmx") }}
```
Comment (clamoriniere) [inline stable/datadog/templates/datadog-yaml-configmap.yaml:38]: good catch

**Comment by clamoriniere** [comment]
@ggmaresca thank you for finding and  fixing this issue.

Could you please:
* bump the version to `2.0.3`
* update `changelog file` with a short description: https://github.com/helm/charts/blob/master/stable/datadog/CHANGELOG.md

File: stable/datadog/templates/container-trace-agent.yaml (hunk 1 above)
Comment (clamoriniere) [inline stable/datadog/templates/container-trace-agent.yaml:18]: which agent version are you using? 
I think the default value is now `true` by default: https://github.com/DataDog/datadog-agent/blob/1c76b8381a195a0b0f629011a6225e936fe1d37a/pkg/config/config.go#L390

Comment (blairham) [reply stable/datadog/templates/container-trace-agent.yaml:18]: I believe it to address this issue:

```bash
2020-03-07 00:16:32 UTC | TRACE | INFO | (pkg/util/log/log.go:482 in func1) | Loaded configuration: /etc/datadog-agent/datadog.yaml
2020-03-07 00:16:32 UTC | TRACE | INFO | (pkg/trace/agent/run.go:71 in Run) | trace-agent not enabled. Set the environment variable
DD_APM_ENABLED=true or add "apm_config.enabled: true" entry
to your datadog.yaml. Exiting...
```

Comment (blairham) [reply stable/datadog/templates/container-trace-agent.yaml:18]: I can confirm that in 7.18.0-rc.6 it's on by default, however.

**Comment by clamoriniere** [comment]
Hi @ggmaresca 
I think something went wrong when you did the git rebase.
Could you fix it, and I will be happy to merge it.
Regards

**Comment by clamoriniere** [comment]
Sorry I did't saw your new PR.

**Pull request closed without merging**